greek-and-roman-musical-studies.csl

```diff
@@ -0,0 +1,291 @@
+              <text macro="title" quotes="false"/>
+              <number variable="volume" form="numeric" prefix=" " suffix=" "/>
+              <text macro="editor"/>
+              <text macro="translator"/>
+              <text macro="additional-info"/>
+              <choose>
+                <if type="motion_picture" match="any" variable="URL">
+                  <group prefix=". ">
+                    <text macro="access"/>
+                  </group>
+                </if>
+              </choose>
+            </group>
+            <text macro="publisher"/>
+          </group>
+        </if>
+        <else-if type="chapter paper-conference" match="any">
+          <text macro="title" prefix=" " suffix="."/>
+          <group>
+            <group delimiter=", " prefix=" ">
+              <group prefix=" ">
+                <text term="in" text-case="capitalize-first" suffix=": "/>
+                <text macro="container-contributors"/>
+                <text variable="container-title" font-style="italic"/>
+                <number variable="volume" form="numeric" prefix=" " suffix=" "/>
+                <text macro="translator"/>
+                <text macro="additional-info"/>
+              </group>
+              <text macro="publisher"/>
+              <text macro="pages" prefix="pp."/>
+            </group>
+            <text macro="doi"/>
+          </group>
+        </else-if>
+        <else-if type="article-journal article-magazine article-newspaper" match="any">
+          <group prefix=" " delimiter=" ">
+            <group>
+              <text macro="title" suffix=". "/>
+              <text variable="container-title" font-style="italic" suffix=" "/>
+              <text variable="volume" suffix=","/>
+              <text macro="pages" prefix=" pp."/>
+              <text macro="doi"/>
+            </group>
+          </group>
+        </else-if>
+        <else-if type="thesis">
+          <group prefix=" " delimiter=", " suffix=" ">
+            <text macro="title"/>
+            <text variable="genre"/>
+            <text variable="publisher"/>
+          </group>
+        </else-if>
+        <else-if type="webpage">
+          <group prefix=" ">
+            <text macro="title" suffix=" [Online]"/>
+            <text macro="edition"/>
+            <text prefix=" " macro="access"/>
+          </group>
+        </else-if>
+        <else>
+          <group>
+            <text macro="title" prefix=". "/>
+            <text macro="editor" prefix=" "/>
+          </group>
+          <group prefix=" " suffix=".">
+            <text variable="container-title" font-style="italic"/>
+            <group prefix=" ">
+              <text variable="volume"/>
+              <text variable="issue" prefix="(" suffix=")"/>
+            </group>
+            <group prefix=", ">
+              <label variable="locator" form="short"/>
+              <text macro="pages"/>
+            </group>
+            <group prefix=". ">
```

Comment:
remove groups with only one item in them (also, there are two prefixes here; don't rely on the processor to remove unnecessary double spacing)